src/citrine/builders/auto_configure.py

```diff
@@ -0,0 +1,183 @@
+from uuid import UUID
+from typing import Union, Dict
+
+from gemd.entity.link_by_uid import LinkByUID
+from gemd.entity.template.has_property_templates import HasPropertyTemplates
+from gemd.entity.template.has_condition_templates import HasConditionTemplates
+from gemd.entity.template.has_parameter_templates import HasParameterTemplates
+from gemd.util.impl import flatten
+
+from citrine.jobs.waiting import wait_while_validating
+from citrine.informatics.data_sources import GemTableDataSource
+from citrine.informatics.descriptors import FormulationDescriptor
+from citrine.informatics.executions.design_execution import DesignExecution
+from citrine.informatics.scores import Score
+from citrine.informatics.workflows import DesignWorkflow
+
+from citrine.resources.data_concepts import _make_link_by_uid
+from citrine.resources.material_run import MaterialRun
+from citrine.resources.project import Project
+from citrine.resources.table_config import TableBuildAlgorithm
+
+
+def extract_descriptor_keys(
+        *,
+        project: Project,
+        material: Union[MaterialRun, LinkByUID, str, UUID],
+        mode: str = 'SIMPLE',
+        full_history: bool = False
+) -> Dict:
+    """[ALPHA] Extract names of attributes from a provided material history.
+
+    Given a material (or link representation), extracts the names from
+    property, condition, and parameter templates
+    that appear within process and measurement runs
+    throughout the material's history.
+
+    The obtained descriptor keys can be used when constructing
+    objectives for a scoring function in a design workflow.
+
+    Parameters
+    ----------
+    project: Project
+        Project to use when accessing the Citrine Platform
+    material: Union[MaterialRun, LinkByUID, str, UUID]
+        Sample material to extract descriptor keys
+    mode: str
+        Option in {'SIMPLE', 'FORMULATION'} that controls formatting descriptor keys
+        Default: 'SIMPLE'
+    full_history: bool
+        Whether to extract descriptor keys from the full material history,
+        or only the provided (terminal) material
+        Default: True
+
+    Returns
+    -------
+    dict
+        Dictionary of keys of the form
+        {'properties': [...], 'conditions': [...], 'parameters': []}
+
+    """
+    if mode not in {"FORMULATION", "SIMPLE"}:
+        msg = "Called with mode: {}.  Expected 'SIMPLE' or 'FORMULATION'.".format(mode)
+        raise ValueError(msg)
+
+    link = _make_link_by_uid(material)
+    history = project.material_runs.get_history(scope=link.scope, id=link.id)
```

Comment:
`get_history` takes a `Union[MaterialRun, LinkByUID, str, UUID]` as the argument to `id`, and the argument `scope` is deprecated. So just call `project.material_runs.get_history(id=material)`

Reply:
This actually is a small bug on `material_runs.get_history`, which does not have an appropriate default option of `scope=None` to allow for that signature.

I will change, and add the default argument in my changes.

Reply:
Yep that's a bug, thank you